pkg/controller/daemon/update.go

```diff
@@ -187,17 +191,6 @@ func (dsc *DaemonSetsController) cleanupHistory(ds *apps.DaemonSet, old []*apps.
 	return nil
 }
 
-// maxRevision returns the max revision number of the given list of histories
-func maxRevision(histories []*apps.ControllerRevision) int64 {
```

Comment:
I don't mind removing this function, but I would leave it in a separate loop vs combing it with the above loop. There's a lot going on in that loop already, and I don't think this change helps with the clarity of what's going on.